src/plots/cartesian/select.js

```diff
@@ -349,6 +289,303 @@ function prepSelect(e, startX, startY, dragOptions, mode) {
     };
 }
 
+function selectOnClick(evt, gd, xAxes, yAxes, subplot, dragOptions, polygonOutlines) {
+    var hoverData = gd._hoverdata;
+    var clickmode = gd._fullLayout.clickmode;
+    var sendEvents = clickmode.indexOf('event') > -1;
+    var selection = [];
+    var searchTraces;
+    var searchInfo;
+    var currentSelectionDef;
+    var selectionTester;
+    var traceSelection;
+    var thisTracesSelection;
+    var pointOrBinSelected;
+    var subtract;
+    var eventData;
+    var i;
```

Comment:
🐄 we usually combine uninitialized vars multiple per line, like https://github.com/plotly/plotly.js/pull/2944/files#diff-8c5d8f688f2bb41a18d35397e5f58864R54

Reply:
Btw. I quickly looked through ESLint to see if there's an appropriate rule. First, I thought https://eslint.org/docs/rules/one-var would fit our purpose, but IMO it's not.